servers/peerrpc/listener.go

```diff
@@ -9,44 +10,54 @@ import (
 	"google.golang.org/grpc/grpclog"
 )
 
-var (
-	server *grpc.Server
-)
-
 func init() {
 	// The GD2 logger for gRPC
 	grpclog.SetLogger(log.StandardLogger().WithField("module", "gRPC"))
 }
 
-// StartListener is to register all the services and start listening on them
-// TODO: This should be able to listen on multiple listeners
-func StartListener() error {
-	server = grpc.NewServer()
-	registerServices(server)
+// Server is the gRPC server
+// It provides an implementation of github.com/thejerf/suture.Service interface
+type Server struct {
+	server *grpc.Server
+}
 
+// New returns a new peerrpc.Server with registered gRPC services
+func New() *Server {
+	s := &Server{
+		grpc.NewServer(),
+	}
+	registerServices(s.server)
+
+	return s
+}
+
+// Serve starts a gRPC server
+// TODO: This should be able to listen on multiple listeners
+func (s *Server) Serve() {
 	listenAddr := config.GetString("peeraddress")
 
 	l, e := net.Listen("tcp", listenAddr)
 	if e != nil {
 		log.WithField("error", e).Error("net.Listen() error")
-		return e
+		return
 	}
 	log.WithField("ip:port", listenAddr).Info("Registered RPC Listener")
 
-	for s, si := range server.GetServiceInfo() {
+	for svc, si := range s.server.GetServiceInfo() {
 		for _, m := range si.Methods {
 			log.WithFields(log.Fields{
-				"service": s,
+				"service": svc,
 				"method":  m,
 			}).Debug("registered gRPC method")
 		}
 	}
 
-	go server.Serve(l)
-	return nil
+	s.server.Serve(l)
+	return
 }
 
-// StopServer stops the server
-func StopServer() {
-	server.GracefulStop()
+// Stop stops the server
+func (s *Server) Stop() {
+	s.server.GracefulStop()
```

Comment:
Okay. Just checked and `s.server.GracefulStop()` internally closes all listeners and this isn't muxed :+1: